flake.nix

```diff
@@ -0,0 +1,31 @@
+{
+  inputs.nixpkgs.url = "github:nixos/nixpkgs";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+
+  outputs = { self, nixpkgs, flake-utils, ... }@inputs:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+        shellPackages = [
+          "jre"
+          "sbt"
+          "nodejs"
+          "yarn"
+        ];
+      in
+      {
+        devShell = pkgs.mkShell {
+          buildInputs = map (pkgName: pkgs.${pkgName}) shellPackages;
+          welcomeMessage = ''
+            Welcome to the smithy4s Nix shell! 👋
+            Available packages:
+            ${builtins.concatStringsSep "\n" (map (n : "- ${n}") shellPackages)}
+          '';
+
+          shellHook = ''
+            echo "$welcomeMessage"
+          '';
```

Comment:
This is how this shows up. Doesn't work if you use direnv and the shell is cached, but it's something.

<img width="763" alt="image" src="https://user-images.githubusercontent.com/894884/150571785-9af6645f-01a7-4ffe-96bc-23ee90b4ba10.png">